src/include/platform/internal/GenericPlatformManagerImpl_CMSISOS.h

```diff
@@ -0,0 +1,182 @@
+    void _Shutdown(void);
+
+#if CHIP_STACK_LOCK_TRACKING_ENABLED
+    bool _IsChipStackLockedByCurrentThread() const;
+#endif
+
+    CHIP_ERROR _PostBackgroundEvent(const ChipDeviceEvent * event);
+    void _RunBackgroundEventLoop(void);
+    CHIP_ERROR _StartBackgroundEventLoopTask(void);
+    CHIP_ERROR _StopBackgroundEventLoopTask();
+
+    // ===== Methods available to the implementation subclass.
+
+private:
+    // ===== Private members for use by this class only.
+
+    inline ImplClass * Impl() { return static_cast<ImplClass *>(this); }
+
+    static void EventLoopTaskMain(void * pvParameter);
+    uint32_t SyncNextChipTimerHandling();
+    uint32_t mNextTimerBaseTime;
+    uint32_t mNextTimerDurationTicks;
+    std::atomic<bool> mShouldRunEventLoop;
+
+#if defined(CHIP_CONFIG_CMSISOS_USE_STATIC_QUEUE) && CHIP_CONFIG_CMSISOS_USE_STATIC_QUEUE
+    uint8_t mEventQueueBuffer[CHIP_DEVICE_CONFIG_MAX_EVENT_QUEUE_SIZE * sizeof(ChipDeviceEvent)];
+    osMessageQueue_t mEventQueueStruct;
+    const osMessageQueueAttr_t mEventQueueAttr = { .cb_mem  = &mEventQueueStruct,
+                                                   .cb_size = osMessageQueueCbSize,
+                                                   .mq_mem  = mEventQueueBuffer,
+                                                   .mq_size = sizeof(mEventQueueBuffer) };
+
+    const osMessageQueueAttr_t * mEventQueueAttrPtr = &mEventQueueAttr;
+#else
+    // Nothing to configure for queues, Just use this to avoid #ifdef in the class implementation
+    const osMessageQueueAttr_t * mEventQueueAttrPtr = nullptr;
+#endif // CHIP_CONFIG_CMSISOS_USE_STATIC_QUEUE
+
+#if defined(CHIP_CONFIG_CMSISOS_USE_STATIC_TASK) && CHIP_CONFIG_CMSISOS_USE_STATIC_TASK
+    uint8_t mEventLoopStack[CHIP_DEVICE_CONFIG_CHIP_TASK_STACK_SIZE];
+    osThread_t mEventLoopTaskControlBlock;
+#endif // CHIP_CONFIG_CMSISOS_USE_STATIC_TASK
+
+    const osThreadAttr_t mEventLoopTaskAttr = {
+        .name      = CHIP_DEVICE_CONFIG_CHIP_TASK_NAME,
+        .attr_bits = osThreadDetached,
+#if defined(CHIP_CONFIG_CMSISOS_USE_STATIC_TASK) && CHIP_CONFIG_CMSISOS_USE_STATIC_TASK
+        .cb_mem    = &mEventLoopTaskControlBlock,
+        .cb_size   = osThreadCbSize,
+        .stack_mem = mEventLoopStack,
+#endif // CHIP_CONFIG_CMSISOS_USE_STATIC_TASK
+        .stack_size = CHIP_DEVICE_CONFIG_CHIP_TASK_STACK_SIZE,
+        .priority   = CHIP_DEVICE_CONFIG_CHIP_TASK_PRIORITY
+    };
+
+#if defined(CHIP_CONFIG_CMSISOS_USE_STATIC_MUTEX) && CHIP_CONFIG_CMSISOS_USE_STATIC_MUTEX
+    osMutexCbSize uint8_t mMutexControlBlock[osMutexCbSize];
+#endif // CHIP_CONFIG_CMSISOS_USE_STATIC_MUTEX
+    const osMutexAttr_t mChipStackMutexAttr = {
+        .name = "",
+#if defined(CHIP_CONFIG_CMSISOS_USE_STATIC_MUTEX) && CHIP_CONFIG_CMSISOS_USE_STATIC_MUTEX
+        .cb_mem  = &mMutexControlBlock,
+        .cb_size = osMutexCbSize,
+#endif // CHIP_CONFIG_CMSISOS_USE_STATIC_MUTEX
+        .attr_bits = osMutexRecursive | osMutexPrioInherit,
```

Comment:
Do we want it to be recursive? the FreeRTOS version of it is not recursive if i am not mistaken.